library/Icinga/Authentication/PasswordValidator.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/* Icinga Web 2 | (c) 2025 Icinga GmbH | GPLv2+ */
+
+namespace Icinga\Authentication;
+
+use Icinga\Application\Hook\PasswordPolicyHook;
+use Zend_Validate_Abstract;
+
+class PasswordValidator extends Zend_Validate_Abstract
+{
+    /**
+     * @var PasswordPolicyHook|null
+     */
+    private ?PasswordPolicyHook $passwordPolicyObject;
+
+    /**
+     * Constructor
+     *
+     * @param PasswordPolicyHook|null $passwordPolicyObject
+     */
+    public function __construct(?PasswordPolicyHook $passwordPolicyObject = null)
```

Comment:
Let's just require the callers to provide a policy.

```suggestion
    public function __construct(PasswordPolicyHook $passwordPolicyObject)
```